sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/specialized/BlockBlobAPITest.groovy

```diff
@@ -2071,6 +2109,40 @@ class BlockBlobAPITest extends APISpec {
         destinationProperties.getAccessTier() == AccessTier.COOL
     }
 
+    def "Upload from URL source oauth"() {
+        setup:
+        def sourceBlob = cc.getBlobClient(generateBlobName())
+        sourceBlob.upload(data.defaultBinaryData)
+        def oauthHeader = getAuthToken().block()
```

Comment:
I think we can move this `block` inside getAuthToken. I don't think we'll ever chain some reactive flow from this.